### PR TITLE
feat: admin

### PR DESCRIPTION
**Pull request by weronika28**
I created an admin panel - user and category management. Backend and frontend work. Support for adding icons will be added (temporary solution for now)

### CODE REVIEW

File: frontend/src/main/java/pl/pollub/frontend/controller/LoginController.java [hunk 1]
```diff
@@ -126,6 +126,7 @@ private void handleValidLogin(JsonObject response) {
 
         authService.setUser(user);
 
-        screenService.switchTo("home");
+        //screenService.switchTo("home");
+        screenService.switchTo("admin");
```
Comment (atakowiec) [inline frontend/src/main/java/pl/pollub/frontend/controller/LoginController.java:130]: revert this

File: frontend/src/main/java/pl/pollub/frontend/service/AdminService.java [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+public class AdminService {
+
+    @Inject
+    private HttpService httpService;
+
+        public List<User> getAllUsers() {
+            HttpResponse<String> response = httpService.get("/api/admin/users");
+            if (response.statusCode() != 200) {
+                throw new RuntimeException("Błąd pobierania użytkowników: " + response.statusCode());
+            } else {
```
Comment (atakowiec) [inline frontend/src/main/java/pl/pollub/frontend/service/AdminService.java:24]: here remove this else, we dont want to have this nested code

File: frontend/src/main/java/pl/pollub/frontend/service/AdminService.java [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+        Map<String, Object> profileData = new HashMap<>();
+        profileData.put("spendingLimit", limit);
+        String json = JsonUtil.GSON.toJson(profileData);
+        return httpService.put("/api/admin/" + userId + "/limit", json);
+    }
+
+    public HttpResponse<String> updateRole(Long userId, String role) {
+        Map<String, String> profileData = new HashMap<>();
+        profileData.put("role", role);
+        String json = JsonUtil.GSON.toJson(profileData);
```
Comment (atakowiec) [inline frontend/src/main/java/pl/pollub/frontend/service/AdminService.java:90]: here and above you dont have to manually stringify this to json, you can pass a map object into the http service methods

File: frontend/src/main/java/pl/pollub/frontend/service/AdminService.java [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+
+    public HttpResponse<String> deleteUser(Long userId) {
+        return httpService.delete("/api/admin/" + userId);
+    }
+
+    public List<TransactionCategory> getExpenseCategories() {
+        HttpResponse<String> response = httpService.get("categories/expenses");
+        if (response.statusCode() != 200) {
+            throw new RuntimeException("Błąd pobierania kategorii wydatków: " + response.statusCode());
+        } else {
```
Comment (atakowiec) [inline frontend/src/main/java/pl/pollub/frontend/service/AdminService.java:102]: nested code

File: frontend/src/main/java/pl/pollub/frontend/service/AdminService.java [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+                throw new RuntimeException("Błąd przy konwersji JSON do listy kategorii wydatków");
+            }
+        }
+    }
+
+    public List<TransactionCategory> getIncomeCategories() {
+        HttpResponse<String> response = httpService.get("categories/incomes");
+        if (response.statusCode() != 200) {
+            throw new RuntimeException("Błąd pobierania kategorii przychodów: " + response.statusCode());
+        } else {
```
Comment (atakowiec) [inline frontend/src/main/java/pl/pollub/frontend/service/AdminService.java:126]: same, nested code

File: frontend/src/main/java/pl/pollub/frontend/service/AdminService.java [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+        return httpService.post("/categories/expenses", json);
+    }
+
+    public HttpResponse<String> addIncomeCategory(TransactionCategory category) {
+        String json = JsonUtil.GSON.toJson(category);
+        return httpService.post("/categories/incomes", json);
+    }
+
+    public HttpResponse<String> updateExpenseCategory(TransactionCategory category) {
+        String json = JsonUtil.GSON.toJson(category);
```
Comment (atakowiec) [inline frontend/src/main/java/pl/pollub/frontend/service/AdminService.java:157]: same, just pass the category object

**Review by atakowiec — CHANGES_REQUESTED** [review]

**Comment by weronika28** [comment]
done

**Review by atakowiec — APPROVED** [review]
very cool

**Pull request merged**